### PR TITLE
Add type declaration for Parser.assert in parsimmon 1.13

### PR DESCRIPTION
**Pull request by Wilfred**
This was added in parsimmon 1.13:
https://github.com/jneen/parsimmon/blob/master/CHANGELOG.md#1130-2019-07-25c

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## parsimmon ([<kbd>unpkg</kbd>](https://unpkg.com/browse/parsimmon@latest/))
was missing the following properties:
1. fantasy-land/empty
1. fantasy-land/of
1. Binary
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f10990c7e92e8de7e4cbe6644097bf6d4b2292d3
</p>

File: types/parsimmon/index.d.ts [hunk 1]
```diff
@@ -107,6 +107,13 @@ declare namespace Parsimmon {
          * about the error.
          */
         tryParse(input: string): T;
+        /**
+         * Passes the result of `parser` to the function `condition`,
+         * which returns a boolean. If the the condition is false, returns
+         * a failed parse with the given `message`. Else it returns the
+         * original result of `parser`.
+         */
+        assert<U>(condition: (result: U) => boolean, message: string): Parser<U>;
```
Comment (caasi) [inline types/parsimmon/index.d.ts:116]: Type `T` should be enough for the result.

see: https://github.com/jneen/parsimmon/blob/99af8f8d8dd0c1aa8542e8ef930805b295194086/src/parsimmon.js#L893-L897

Comment (Wilfred) [reply types/parsimmon/index.d.ts]: I've updated the PR to just use `T` :)

Comment (caasi) [reply types/parsimmon/index.d.ts]: Thank you!

**Comment by Wilfred** [comment]
@caasi anything else you want here, or is this good to merge?

**Review by caasi — APPROVED** [review]

**Comment by caasi** [comment]
@Wilfred Sorry I forget to approve your PR. Thank you for your patient.

**Comment by Wilfred** [comment]
Ready to merge

**Pull request merged**